mplc/scenario.py

```diff
@@ -724,66 +734,16 @@ def compute_batch_sizes(self):
-                )
-                partner.shuffle_labels(1.0)
-            elif self.corrupted_datasets[partner_index] == 'permuted':
-                logger.debug(
-                    f"  ... Corrupting (by permuting the labels ) the whole dataset of partner #{partner.id}."
-                    f"Permutation matrix available"
-                )
-                partner.permute_labels(1.0)
-            elif self.corrupted_datasets[partner_index] == 'random':
-                logger.debug(
-                    f"  ... Corrupting (by randomizing the labels ) the whole dataset of partner #{partner.id}."
-                    f"Dirichlet distribution matrix available"
-                )
-                partner.random_labels(1.0)
-            elif self.corrupted_datasets[partner_index][0] == "corrupted":
-                logger.debug(
-                    f"   ... Corrupting (by offsetting labels) {self.corrupted_datasets[partner_index][1] * 100} \
-                            percent of the data of partner #{partner.id}"
-                )
-                partner.corrupt_labels(self.corrupted_datasets[partner_index][1])
-            elif self.corrupted_datasets[partner_index][0] == "shuffled":
-                logger.debug(
-                    f"   ... Corrupting (by shuffling labels) {self.corrupted_datasets[partner_index][1] * 100} \
-                            percent of the data of partner #{partner.id}"
-                )
-                partner.shuffle_labels(self.corrupted_datasets[partner_index][1])
-            elif self.corrupted_datasets[partner_index][0] == 'permuted':
-                logger.debug(
-                    f"  ... Corrupting (by permuting the labels ) {self.corrupted_datasets[partner_index][1] * 100} "
-                    f"percent of the data of partner #{partner.id}."
-                    f"Permutation matrix available"
-                )
-                partner.permute_labels(self.corrupted_datasets[partner_index][1])
-            elif self.corrupted_datasets[partner_index][0] == 'random':
-                logger.debug(
-                    f"  ... Corrupting (by randomizing the labels ) {self.corrupted_datasets[partner_index][1] * 100} "
-                    f"percent of the data of partner #{partner.id}."
-                    f"Dirichlet distribution matrix available"
-                )
-                partner.random_labels(self.corrupted_datasets[partner_index][1])
-            elif self.corrupted_datasets[partner_index] == "not_corrupted":
-                pass
-            else:
-                logger.debug("Unexpected label of corruption, no corruption performed!")
-
-            logger.debug(f"   Partner #{partner.id}: done.")
+        """perform corruption on partner if needed"""
+        for partner in self.partners_list:
+            if partner.corruption_method == "duplication":
+                if not partner.duplicated_partner_id:
```

Comment:
I don't understand this case.

Reply:
If the 'duplication' method is selected and the partner id to copy is not provided.